### PR TITLE
MI-404: Initialise survey on plugin startup

### PR DESCRIPTION
**Pull request by chetanyakan**

### CODE REVIEW

File: server/model/survey.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+const (
+	TypeSurvey           ValueType = "survey"
+	TypeSurveyResponse   ValueType = "survey_response"
+	TypeMeetingMetadata  ValueType = "meeting_metadata"
+	TypeLatestSurveyInfo ValueType = "latest_survey_info"
+)
+
+type QuestionType string
+
+const (
```
Comment (harshilsharma63) [inline server/model/survey.go]: Can we club the two `const` sections together?

File: server/model/survey.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
+package model
+
+import (
+	"encoding/json"
+
+	serverModel "github.com/mattermost/mattermost-server/model"
+)
+
+type ValueType string
```
Comment (harshilsharma63) [inline server/model/survey.go]: Does this type provide any benefit over directly using string?

File: server/model/survey.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	MeetingID     string
+	SurveyID      string
+	SurveyVersion int
+	UserResponded map[string]bool
+}
+
+// LatestSurveyInfo stores the latest version information for a survey
+type LatestSurveyInfo struct {
+	Type    ValueType `json:"type"`
+	ID      string    `json:"id"`
```
Comment (harshilsharma63) [inline server/model/survey.go]: Is this the survey ID?

File: server/store/kvstore/survey.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import (
+	"strconv"
+
+	"github.com/Brightscout/mattermost-plugin-survey/server/config"
+	"github.com/Brightscout/mattermost-plugin-survey/server/model"
+	"github.com/Brightscout/mattermost-plugin-survey/server/store"
+)
+
+// Store is the implementation for the interface to interact with the KV Store.
+type Store struct {
```
Comment (harshilsharma63) [inline server/store/kvstore/survey.go]: Can we change it to something else which reflects which implementation of `Store` this is?

File: server/store/store.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package store
+
+import (
+	"github.com/Brightscout/mattermost-plugin-survey/server/model"
+)
+
+// Store allows the interaction with some kind of store.
+type Store interface {
+	GetLatestSurveyInfo(id string) (*model.LatestSurveyInfo, error)
```
Comment (harshilsharma63) [inline server/store/store.go:9]: I don't think all of these functions are generic, `Store` functions which every implementation of `Store` would need. Let's think about if we can refactor the interface.

**Review by harshilsharma63 — CHANGES_REQUESTED** [review]

File: server/model/survey.go (hunk 1 above)
Comment (chetanyakan) [reply server/model/survey.go]: Done.

File: server/model/survey.go (hunk 2 above)
Comment (chetanyakan) [reply server/model/survey.go]: No, it does not. Updated.

File: server/model/survey.go (hunk 3 above)
Comment (chetanyakan) [reply server/model/survey.go]: Yes. Updated here: https://github.com/Brightscout/mattermost-plugin-survey/pull/4/commits/aa085e299f4b88624aa2a2185c8e63a94e409e07

File: server/store/kvstore/survey.go (hunk 4 above)
Comment (chetanyakan) [reply server/store/kvstore/survey.go]: Updated here: https://github.com/Brightscout/mattermost-plugin-survey/pull/4/commits/66b70db80c77e437bee530964789bc08a0872608

File: server/store/store.go (hunk 5 above)
Comment (chetanyakan) [reply server/store/store.go:9]: 1. As discussed, for other Survey implementations, an interface for the business logic layer would be required.
2. Updated the interface name to `SurveyStore` here: https://github.com/Brightscout/mattermost-plugin-survey/pull/4/commits/66b70db80c77e437bee530964789bc08a0872608

**Review by ghost — APPROVED** [review]

**Pull request merged**